openedx/features/idea/api_views.py

```diff
@@ -0,0 +1,26 @@
+from django.http import JsonResponse
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+from openedx.features.idea.models import Idea
+
+
+class FavoriteAPIView(APIView):
+    """
+    FavoriteAPIView is used t toggle favorite idea for the user
+    """
+    authentication_classes = (SessionAuthentication, BasicAuthentication)
```

Comment:
@ahmed-faraz46 do we need to allow inactive users to toggle favorites?If so these auth classes will throw 401 error in case user is in active

Reply:
@monofal ideally it doesn't make sense if inactive user can make activities

Reply:
yes but in philu inactive user is allowed to perform various operations.

Reply:
@monofal good point lets discuss in standup tomorrow.